signer/xpi/x509.go

```diff
@@ -78,6 +73,16 @@ func (s *PKCS7Signer) MakeEndEntity(cn string) (eeCert *x509.Certificate, eeKey
 		NotAfter:           time.Now().Add(8760 * time.Hour), // one year
 		SignatureAlgorithm: s.issuerCert.SignatureAlgorithm,
 	}
+
+	// having a dnsname or an email address field is required to comply with webpki
+	// requirement due to the intermediates having constraints
+	// see https://github.com/golang/go/issues/24151
+	idEmailRegex := regexp.MustCompile("^[a-zA-Z0-9._~-]+@[a-zA-Z0-9._~-]+")
```

Comment:
Not really a fan of regexes, maybe https://golang.org/pkg/net/mail/#AddressParser.Parse instead?